packages/catalog-process/src/services/catalogService.ts

```diff
@@ -91,6 +105,26 @@ const assertRequesterAllowed = (
   }
 };
 
+function assertIsDraftEservice(eservice: EService): void {
+  if (eservice.descriptors.some((d) => d.state !== descriptorState.draft)) {
```

Comment:
an eservice is considered in draft state also if it has no descriptors

Reply:
This should cover the case: if it has no descriptors, `some(...)` is `false` - there are no descriptors that match the condition, so it will not throw `eserviceNotInDraftState`